bin/api_diff.ml

```diff
@@ -40,8 +53,20 @@ let mode ~reference ~current ~main_module ~unwrapped =
         "Either --main-module or --unwrapped must be provided when diffing \
          entire libraries."
 
-let run (`Main_module main_module) (`Unwrapped_library unwrapped)
-    (`Ref_cmi reference) (`Current_cmi current) =
+let print_diff diff word_diff =
+  let open CCResult.Infix in
+  let text_diff = Api_watch.Text_diff.from_diff diff in
+  let* display_mode = display_mode word_diff in
```

Comment:
This can be moved outside of this function as it only needs to be computed once for the whole program. You can make `print_diff` take the `display_mode` directly.

This allow to use `List.iter` too, without poluting it with the `result`!